src/ChipInput.js

```diff
@@ -344,8 +344,9 @@ class ChipInput extends React.Component {
       }
     } else {
       if (this.props.dataSourceConfig) {
-        const chips = this.state.chips.filter((c) => c[this.props.dataSourceConfig.value] !== chip)
-        if (chips.length !== this.state.chips.length) {
+        const chips = this.state.chips;
+        let changed = chips.splice(i,1) // remove the chip at index i
```

Comment:
This mutates `chips`, which is a property of `this.state` and thus should never be changed, doesn't it? We need to create a _copy_ of the array without the element that should be removed.